packages/flutter/lib/src/services/binding.dart

```diff
@@ -415,6 +416,12 @@ mixin ServicesBinding on BindingBase, SchedulerBinding {
   void setSystemUiChangeCallback(SystemUiChangeCallback? callback) {
     _systemUiChangeCallback = callback;
   }
+
+  /// Alert the engine that the binding is registered. This instructs the engine to
+  /// register its top level window handler on Windows.
+  Future<void> enableApplicationLifecycle() async {
+    SystemChannels.platform.invokeMethod('System.enableApplicationLifecycle');
```

Comment:
Loïc and I discussed making this `"System.initializationComplete"` so that this wouldn't be so tied to app lifecycle, but could be used as a more generic signal.  That's what I currently have in my implementations for macOS and Linux.